rules/models/changes/mapping.go

```diff
@@ -0,0 +1,60 @@
+package changes
+
+import "github.com/m3db/m3metrics/rules/models"
+
+// MappingRuleChange is a mapping rule diff.
+type MappingRuleChange struct {
+	Op       UpdateOp            `json:"updateOp"`
+	RuleID   string              `json:"ruleID,omitempty"`
```

Comment:
Should the rule ID be a pointer as well since add operations don't have a rule ID set?

Reply:
I was taking a look at this. When encoding a JSON struct that has this filed with an empty string the `omitempty` struct tag will remove `ruleID` from the resulting JSON object. When hydrating an instance of the struct from a JSON byte blob this field will be included as an empty string. It seems like having a pointer to a string may be more prone to panics, but maybe we want to fail fast in this situation. I can change this to a pointer if we think its a good idea but as far as the JSON encoding goes this is valid.